src/seedu/addressbook/data/person/Address.java

```diff
@@ -37,22 +49,101 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return block + ", " + street + ", " + unit + ", " + postalCode;
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && this.block.equals(((Address) other).block)
+                && this.street.equals(((Address) other).street)
+                && this.unit.equals(((Address) other).unit)
+                && this.postalCode.equals(((Address) other).postalCode)); // state check
```

Comment:
Have you tried running the tests? What if you entered the same details twice? If the addresses are identical, are they detected to be equal?